source/usage-examples/insertOne.txt

```diff
@@ -3,3 +3,52 @@ Insert a Document
 =================
 
 .. default-domain:: mongodb
+
+You can insert a document into a collection by using the ``InsertOne()``
+method.
+
+The following example creates a new document containing a ``title`` and ``text``
+field and inserts it into the ``haikus`` collection of the ``insertDB`` database:
+
+.. note:: Non-existent Database and Collections
+
+    The driver automatically creates the necessary database and/or collection
+    when you perform a write operation against them, if they don't exist.
+
+
+.. include:: /includes/usage-examples/run-example-tip.rst
+
+.. literalinclude:: /includes/usage-examples/code-snippets/insertOne.go
+   :start-after: begin insertOne
+   :end-before: end insertOne
+   :emphasize-lines: 4
+   :language: go
+   :dedent:
+
+Click `here <{+example+}/insertOne.go>`__ to see a fully runnable example.
+
+Expected Result
+---------------
+
+After you run the code example, a new document with an ``_id`` that matches
+the output will be present:
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     _id: ObjectId("..."),
+     title: 'An Insert Haiku',
+     text: 'No bytes, no problem. Inserting a document. In MongoDB'
```

Comment:
```suggestion
     title: 'Record of a Shriveled Datum',
     text: 'No bytes, no problem. Inserting a document. In MongoDB',
     style:'haiku'
```

Nit: Can title be more haiku like? Got inspiration from [this wiki](https://en.wikipedia.org/wiki/Matsuo_Bash%C5%8D#List_of_works)